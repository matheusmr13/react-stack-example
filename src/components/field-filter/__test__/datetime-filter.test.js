import React from 'react';
import DatetimeFilter from 'components/field-filter/datetime-filter';
import { shallow } from 'enzyme';
import moment from 'moment';

describe('DatetimeFilter component', () => {
	it('should render properly', () => {
		const datetimeFilter = shallow(<DatetimeFilter spec={{ id: 'date', name: 'Date field' }} />);
		expect(datetimeFilter).toMatchSnapshot();
	});
	it('should merge date and time', () => {
		let lastChangedDate;

		const onChange = (value) => {
			lastChangedDate = value;
		};
		const datetimeFilter = shallow(<DatetimeFilter spec={{ id: 'date', name: 'Date field' }} onChange={onChange} />);
		expect(lastChangedDate).toBeUndefined();

		const dateMoment = moment({
			years: 2018,
			month: 10,
			date: 15
		});
		const hourMoment = moment({
			minutes: 40,
			hours: 10
		});
		datetimeFilter.instance().onChange('date', dateMoment.toDate());
		datetimeFilter.instance().onChange('time', hourMoment.toDate());

		expect(lastChangedDate).not.toBeUndefined();
		expect(lastChangedDate).toBe('2018-11-15T10:40:00-02:00');
	});
});
