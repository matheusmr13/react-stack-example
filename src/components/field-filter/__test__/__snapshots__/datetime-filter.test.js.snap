// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatetimeFilter component should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatetimeFilter
    onChange={[Function]}
    spec={
        Object {
            "id": "date",
            "name": "Date field",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
                    Object {
                              "display": "inline-block",
                              "marginRight": "8px",
                              "width": "calc(60% - 8px)",
                            }
          }
>
          <DatePicker
                    autoOk={false}
                    container="dialog"
                    disableYearSelection={false}
                    disabled={false}
                    firstDayOfWeek={1}
                    fullWidth={true}
                    hideCalendarDate={false}
                    hintText="Date field"
                    onChange={[Function]}
                    openToYearSelection={false}
                    style={Object {}}
          />
</div>,
        <div
          style={
                    Object {
                              "display": "inline-block",
                              "width": "40%",
                            }
          }
>
          <TimePicker
                    autoOk={false}
                    cancelLabel="Cancel"
                    defaultTime={null}
                    disabled={false}
                    format="ampm"
                    fullWidth={true}
                    hintText="00:00"
                    minutesStep={1}
                    okLabel="OK"
                    onChange={[Function]}
                    pedantic={false}
                    style={Object {}}
                    value={null}
          />
</div>,
      ],
      "style": Object {
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <DatePicker
            autoOk={false}
            container="dialog"
            disableYearSelection={false}
            disabled={false}
            firstDayOfWeek={1}
            fullWidth={true}
            hideCalendarDate={false}
            hintText="Date field"
            onChange={[Function]}
            openToYearSelection={false}
            style={Object {}}
/>,
          "style": Object {
            "display": "inline-block",
            "marginRight": "8px",
            "width": "calc(60% - 8px)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoOk": false,
            "container": "dialog",
            "disableYearSelection": false,
            "disabled": false,
            "firstDayOfWeek": 1,
            "fullWidth": true,
            "hideCalendarDate": false,
            "hintText": "Date field",
            "onChange": [Function],
            "openToYearSelection": false,
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TimePicker
            autoOk={false}
            cancelLabel="Cancel"
            defaultTime={null}
            disabled={false}
            format="ampm"
            fullWidth={true}
            hintText="00:00"
            minutesStep={1}
            okLabel="OK"
            onChange={[Function]}
            pedantic={false}
            style={Object {}}
            value={null}
/>,
          "style": Object {
            "display": "inline-block",
            "width": "40%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoOk": false,
            "cancelLabel": "Cancel",
            "defaultTime": null,
            "disabled": false,
            "format": "ampm",
            "fullWidth": true,
            "hintText": "00:00",
            "minutesStep": 1,
            "okLabel": "OK",
            "onChange": [Function],
            "pedantic": false,
            "style": Object {},
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
                        Object {
                                    "display": "inline-block",
                                    "marginRight": "8px",
                                    "width": "calc(60% - 8px)",
                                  }
            }
>
            <DatePicker
                        autoOk={false}
                        container="dialog"
                        disableYearSelection={false}
                        disabled={false}
                        firstDayOfWeek={1}
                        fullWidth={true}
                        hideCalendarDate={false}
                        hintText="Date field"
                        onChange={[Function]}
                        openToYearSelection={false}
                        style={Object {}}
            />
</div>,
          <div
            style={
                        Object {
                                    "display": "inline-block",
                                    "width": "40%",
                                  }
            }
>
            <TimePicker
                        autoOk={false}
                        cancelLabel="Cancel"
                        defaultTime={null}
                        disabled={false}
                        format="ampm"
                        fullWidth={true}
                        hintText="00:00"
                        minutesStep={1}
                        okLabel="OK"
                        onChange={[Function]}
                        pedantic={false}
                        style={Object {}}
                        value={null}
            />
</div>,
        ],
        "style": Object {
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <DatePicker
              autoOk={false}
              container="dialog"
              disableYearSelection={false}
              disabled={false}
              firstDayOfWeek={1}
              fullWidth={true}
              hideCalendarDate={false}
              hintText="Date field"
              onChange={[Function]}
              openToYearSelection={false}
              style={Object {}}
/>,
            "style": Object {
              "display": "inline-block",
              "marginRight": "8px",
              "width": "calc(60% - 8px)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoOk": false,
              "container": "dialog",
              "disableYearSelection": false,
              "disabled": false,
              "firstDayOfWeek": 1,
              "fullWidth": true,
              "hideCalendarDate": false,
              "hintText": "Date field",
              "onChange": [Function],
              "openToYearSelection": false,
              "style": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TimePicker
              autoOk={false}
              cancelLabel="Cancel"
              defaultTime={null}
              disabled={false}
              format="ampm"
              fullWidth={true}
              hintText="00:00"
              minutesStep={1}
              okLabel="OK"
              onChange={[Function]}
              pedantic={false}
              style={Object {}}
              value={null}
/>,
            "style": Object {
              "display": "inline-block",
              "width": "40%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoOk": false,
              "cancelLabel": "Cancel",
              "defaultTime": null,
              "disabled": false,
              "format": "ampm",
              "fullWidth": true,
              "hintText": "00:00",
              "minutesStep": 1,
              "okLabel": "OK",
              "onChange": [Function],
              "pedantic": false,
              "style": Object {},
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
